bezier-rs/lib/src/lib.rs

```diff
@@ -721,6 +699,141 @@ impl Bezier {
 		});
 		result
 	}
+
+	/// Approximate a bezier curve with circular arcs.
+	/// The algorithm can be customized using the [ArcsOptions] structure.
+	pub fn arcs(&self, arcs_options: ArcsOptions) -> Vec<CircleArc> {
+		let ArcsOptions { maximize_arcs, error, max_iterations } = arcs_options;
+		if maximize_arcs {
+			return self.arcs_helper(0., 1., error, max_iterations);
+		}
+		let mut extrema: Vec<f64> = self.local_extrema().into_iter().flatten().collect::<Vec<f64>>();
+		extrema.append(&mut vec![0., 1.]);
+		extrema.dedup();
+		extrema.sort_by(|ex1, ex2| ex1.partial_cmp(ex2).unwrap());
+
+		let mut t1 = extrema[0];
+		extrema
+			.iter()
+			.skip(1)
+			.flat_map(|&t2| {
+				let arcs = self.arcs_helper(t1, t2, error, max_iterations);
+				println!("t1, t2: {}, {}", t1, t2);
+				t1 = t2;
+				arcs
+			})
+			.collect::<Vec<CircleArc>>()
+	}
+
+	/// Implements an algorithm that approximates a bezier curve with circular arcs.
+	/// This algorithm uses a method akin to binary search to find an arc that approximates a maximal segment of the curve.
+	/// Once a maximal arc has been found for a sub-segment of the curve, the algorithm continues by starting again at the end of the previous approximation.
+	/// More details can be found in the [Approximating a Bezier curve with circular arcs](https://pomax.github.io/bezierinfo/#arcapproximation) section of Pomax's bezier curve primer.
+	fn arcs_helper(&self, local_low: f64, local_high: f64, error: f64, max_iterations: i32) -> Vec<CircleArc> {
+		let mut low = local_low;
+		let mut middle = local_low + (local_high - local_low) / 2.;
+		let mut high = local_high;
+
+		let mut previous_high = local_high;
+
+		let mut iterations = 0;
+
+		let mut previous_arc = CircleArc::default();
+		let mut was_previous_good = false;
+
+		let mut arcs = Vec::new();
+
+		// Find the arc
+		'iterate_curve: while low < local_high {
+			'find_good_segment: loop {
```

Comment:
I was told once that having to label your loops is an indication that your function might be too complex. You can always avoid having to do this by just creating a helper function, and then `return` becomes the equivalent of the outer `break`.